start_admin_sshd.sh

```diff
@@ -98,6 +99,7 @@ for key_alg in rsa ecdsa ed25519; do
   # If both of the keys exist, don't overwrite them
   if [[ -s "${SSH_HOST_KEY_DIR}/ssh_host_${key_alg}_key" ]] \
   && [[ -s "${SSH_HOST_KEY_DIR}/ssh_host_${key_alg}_key.pub" ]]; then
+    ln -sf "${SSH_HOST_KEY_DIR}/ssh_host_${key_alg}_key.pub" "${SSHD_CONFIG_DIR}/ssh_host_${key_alg}_key.pub"
```

Comment:
I'm not sure I understand this - why are we symlinking keys there?

Reply:
Actually, it is a kind of trick - as you know, SSH_HOST_KEY_DIR is a permanent while SSHD_CONFIG_DIR (/etc/ssh) is not. Therefore, to make SSHD_CONFIG_DIR persistent across admin container reboot, I just do make symbolic links.

The reason why SSHD_CONFIG_DIR should be persistent, is that the eic_harvest_hostkeys scripts refers /etc/ssh to read host keys, and the path is hard-coded on the script.
https://github.com/aws/aws-ec2-instance-connect-config/blob/master/src/bin/eic_harvest_hostkeys#L82

Reply:
I wonder if it would be beneficial to make the whole `/etc/ssh/` directory a symlink to `/.bottlerocket/host-containers/current/etc/ssh/` as opposed to individual files.

Reply:
It sounds good, but I have a question: sshd_config file is also in the /etc/ssh/ directory, and the file copy done at the Dockerfile build time (https://github.com/bottlerocket-os/bottlerocket-admin-container/blob/develop/Dockerfile#L62)
Should I change the Dockerfile to copy the sshd_config to `/etc/ssh-template/` and `ln -s ... && mv /etc/ssh-template/sshd_config /etc/ssh/sshd_config` in this start script?

Reply:
I tried it out yesterday and I'm not sure it's worth the hassle. I'm fine with your current symlink approach 😄